cli/src/bench.rs

```diff
@@ -37,11 +37,29 @@ pub fn handle(
     let model =
         params.tract_model.downcast_ref::<TypedModel>().context("Can only bench TypedModel")?;
     let inputs = tract_libcli::tensor::retrieve_or_make_inputs(model, &run_params)?.remove(0);
-    let plan = SimplePlan::new_with_options(model, &plan_options)?;
 
     limits.warmup(model, &inputs)?;
 
-    let mut state = SimpleState::new(plan)?;
+    let mut state = {
+        #[cfg(any(target_os = "macos", target_os = "ios"))]
+        {
```

Comment:
should this only happen with `--metal` ?

Reply:
Good catch! Creating a MetalSessionHandler from a non-Metal plan is not really an issue, as it will just do nothing and the model will execute as normal. But for user readability I'll add a check on the flag